src/core/type/type_info.cc

```diff
@@ -181,8 +181,17 @@ std::unique_ptr<Type> string_type() { return std::make_unique<StringType>(); }
 std::unique_ptr<Type> fixed_array_type(std::unique_ptr<Type> element_type,
                                        uint32_t N) noexcept(false)
 {
-  return std::make_unique<FixedArrayType>(
-    Runtime::get_runtime()->get_type_uid(), std::move(element_type), N);
+  // We use UIDs of the following format for "common" fixed array types
+  //    1B            1B
+  // +--------+-------------------+
+  // | length | element type code |
+  // +--------+-------------------+
+  auto generate_uid = [](const Type& elem_type, uint32_t N) {
```

Comment:
Is there a need for a lambda here?

Reply:
It's just a stylistic choice. I've been using a lambda whenever the logic in a scope gets complicated enough to warrant a descriptive name.

Reply:
also, lambdas like this will be just compiled away, so no runtime overhead would exist.